lib/sprout/src/main/kotlin/org/partiql/sprout/generator/target/kotlin/poems/KotlinBuilderPoem.kt

```diff
@@ -30,14 +30,27 @@ class KotlinBuilderPoem(symbols: KotlinSymbols) : KotlinPoem(symbols) {
 
     private val builderPackageName = "${symbols.rootPackage}.builder"
 
+    // Default unique id provider
+    // open val id: () -> Int = run { ... }
+    private val idProviderType = LambdaTypeName.get(returnType = Int::class.asTypeName())
+    private val idProvider = PropertySpec.builder("_id", idProviderType)
+        .addModifiers(KModifier.PRIVATE)
+        .initializer("run { var i = 1; { i++ } }")
```

Comment:
There are some issues with this actually. Keeping state like this in the abstract class would cause weird behavior with the singelton default factory. I'm convinced we want the singleton default factory, so how do we avoid keeping state there? 

I think a stateful version of the DEFAULT factory makes sense.